src/httpfetch.cpp

```diff
@@ -766,6 +767,27 @@ void httpfetch_sync(const HTTPFetchRequest &fetch_request,
 	fetch_result = *ongoing.complete(res);
 }
 
+bool httpfetch_sync_interruptible(const HTTPFetchRequest &fetch_request,
+		HTTPFetchResult &fetch_result, long interval)
+{
+	if (Thread *thread = Thread::getCurrentThread()) {
+		HTTPFetchRequest req = fetch_request;
+		req.caller = httpfetch_caller_alloc_secure();
+		httpfetch_async(req);
+		do {
+			if (thread->stopRequested()) {
+				httpfetch_caller_free(req.caller);
```

Comment:
afaict `httpfetch_caller_free` handles all cases correctly so the `FreeThread` from the previous PR was indeed not needed